### PR TITLE
Support all fields request

### PR DESCRIPTION
**Pull request by itye-msft**

### CODE REVIEW

**Comment by itye-msft** [comment]
@omri374 and @eladiw  pls review the change and let's merge to dev

File: src/template.proto [hunk 1]
```diff
@@ -14,6 +14,8 @@ message AnalyzeTemplate {
     string modifiedTime = 4;
     // Language code, in ISO-639 format, https://en.wikipedia.org/wiki/ISO_639-1
     string languageCode = 5;
+    // Optional flag to request all existing entities. if true, "fields" must be empty.
+    bool allEntities = 6;
```
Comment (limorl) [inline src/template.proto]: Inconsistent terminology.
Either change fields to entities or allEntities to allFields.
I prefer the first option.

Comment (itye-msft) [reply src/template.proto]: @limorl , I fear that renaming fields to entities will exceed the time allocated for this task.
How about to add another task to rename fields to entities, all over the project?

Comment (limorl) [reply src/template.proto]: We cannot have every task causing the creation of another task with development depth growing each sprint. This is a minor task, closing it correctly will probably add a few hours only. 
so please fix according to my comment in one of the two provided options.

Comment (itye-msft) [reply src/template.proto]: @limorl During the code writing I noticed some methods converting fields to entities. Therefore, for now, the name is renamed to allFields to maintain the consistency in the analyzer code.

Comment (limorl) [reply src/template.proto]: Can you also change the field numbering and move allFields to be the one following 'fields'?

Comment (omri374) [reply src/template.proto]: @itye-msft please fix following Limor's suggestion and let's merge this. (move allFields to be the one following fields)

**Comment by itye-msft** [comment]
@omri374 I have changed the order as per request. please review and approve and merge. thanks

**Pull request merged**